### PR TITLE
Feature/saa 462 add transfers to unlock events

### PR DESCRIPTION
**Pull request by matt-crowson-moj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/ScheduledEventService.kt [hunk 1]
```diff
@@ -75,51 +78,68 @@ class ScheduledEventService(
...
       if (!prisonRolledOut.active) {
-        prisonApiClient.getScheduledActivities(bookingId, dateRange)
+        prisonApiClient.getScheduledActivities(prisoner.first, dateRange)
       } else {
         Mono.just(emptyList())
       },
+      fetchPrisonApiExternalTransfersIfRangeIncludesToday(prisonRolledOut, prisoner.second, dateRange),
     )
 
+  private fun fetchPrisonApiExternalTransfersIfRangeIncludesToday(
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/ScheduledEventService.kt:98]: I think the logic is right in here @timharrison-moj but please feel free to correct.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/ScheduledEventService.kt [hunk 2]
```diff
@@ -145,23 +166,65 @@ class ScheduledEventService(
...
+      Mono.just(emptyList())
+    }
+
+  private fun PrisonerSchedule.redacted() = this.copy(
+    comment = null,
+    locationCode = null,
+    locationId = null,
+    eventLocationId = null,
+    eventLocation = null,
+    eventDescription = "",
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/ScheduledEventService.kt:225]: @timharrison-moj 

Should this be setting/overriding the start end end times to match that of the AM/PM sessions?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/model/PrisonerScheduledEvents.kt [hunk 3]
```diff
@@ -22,14 +22,17 @@ data class PrisonerScheduledEvents(
   val endDate: LocalDate?,
 
   @Schema(description = "A list of scheduled appointments for this prisoner in this date range")
-  var appointments: List<ScheduledEvent>? = null,
+  val appointments: List<ScheduledEvent>? = null,
```
Comment (timharrison-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/model/PrisonerScheduledEvents.kt:26]: Good shout on the val/var difference. I was wondering if these need to be nullable at all? Could they be initialised to `emptyList()`?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/ScheduledEventService.kt (hunk 1 above)
Comment (timharrison-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/ScheduledEventService.kt:98]: Looks fine to me.

**Review by timharrison-moj — APPROVED** [review]
LGTM.

**Pull request merged**